pkg/kubelet/status/manager.go

```diff
@@ -323,8 +323,14 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 	// TODO: make me easier to express from client code
 	pod, err := m.kubeClient.Pods(status.podNamespace).Get(status.podName)
 	if errors.IsNotFound(err) {
-		glog.V(3).Infof("Pod %q (%s) was deleted on the server", status.podName, uid)
-		m.deletePodStatus(uid)
+		glog.V(3).Infof("Pod %q (%s) does not exist on the server", status.podName, uid)
+		// Note that we do not delete the pod status because the pod to update may
+		// be a static pod whose mirror pod is in the process of recreation.
+		// Deleting the status without clearing the corresponding
+		// apiStatusVersions entry may lead to wrong versioning and stale status.
```

Comment:
I'm not sure I follow this logic... If the mirror pod is being recreated, it will get a new UID. Now that apiStatusVersions is keyed off mirror pod UID (when applicable), the new mirror pod should effectively see a "cleared' apiStatusVersion.

I'm not arguing that we should delete the pod status (that has lots of side effects now), but I'm not sure I agree with the comment.